core/src/mindustry/world/meta/values/WeaponListValue.java

```diff
@@ -0,0 +1,57 @@
+package mindustry.world.meta.values;
+
+import arc.*;
+import arc.util.*;
+import arc.scene.ui.layout.*;
+import arc.struct.*;
+import mindustry.gen.*;
+import mindustry.type.*;
+import mindustry.ui.*;
+import mindustry.world.meta.*;
+
+import static mindustry.Vars.*;
+
+public class WeaponListValue implements StatValue{
+    private final Seq<Weapon> weapons;
+    private final UnitType unit;
+
+    public WeaponListValue(UnitType unit, Seq<Weapon> weapons){
+        this.weapons = weapons;
+        this.unit = unit;
+    }
+
+    @Override
+    public void display(Table table){
+        table.row();
+        for(int i = 0;i < weapons.size;i ++){
+            Weapon weapon = weapons.get(i);
+
+            if(weapon.flipSprite){
+                //fliped weapons are not given stats
+                continue;
+            }
+
+            if(weapon.outlineRegion.found()){
+                table.image(weapon.outlineRegion).size(10 * 8).right().top();
```

Comment:
Use `.scaling(Scaling.fit)` for weapon images.

Reply:
Added